api/api.yaml

```diff
@@ -1788,6 +1788,114 @@ paths:
               schema:
                 $ref: "../problem/problem.yaml#/components/schemas/ApiProblem"
       x-codegen-request-body-name: generateConnectionParams
+  /projects/{projectIdOrName}/connections/connections/{connectionId}:regenerate:
+    patch:
+      summary: Update an existing connection (only valid for providers with auth types which are not OAuth2 Authorization Code)
+      operationId: regenerateConnection
+      tags: ["Connection"]
+      parameters:
+        - name: projectIdOrName
+          in: path
+          required: true
+          schema:
+            type: string
+        - name: connectionId
+          in: path
+          required: true
+          schema:
+            type: string
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              required:
+                - updateMask
+              properties:
+                updateMask:
```

Comment:
typically for an API endpoint that has an `updateMask`, there should be a top-level object containing possible fields to be updated. (The top level object should be called something like "ConnectionRequest" in this case because it's not the same schema as a Connection). (See [UpdateProviderApp](https://github.com/amp-labs/openapi/blob/bc423f719ac2ad5f80db1167a8c5a4a1e5de14a3/api/api.yaml#L426) for an example, there's a ProviderApp top-level object in the request) By following this format, then the endpoint can make use of all the update mask helper methods we already have.